structuralcodes/sections/specific.py

```diff
@@ -0,0 +1,446 @@
+""" Specific class section implementation"""
+
+from __future__ import annotations
+
+import typing as t
+import structuralcodes.core._section_results as s_res
+from shapely import Polygon
+from structuralcodes.sections._generic import GenericSection
+from structuralcodes.geometry import SurfaceGeometry
+from structuralcodes.materials.concrete import Concrete
+
+class RectangularSection(GenericSection): 
+    """This is the specific implementation of a rectangular class section.
+
+    The section is a 2D geometry where Y axis is horizontal while Z axis is
+    vertical.
+
+    The moments and curvatures around Y and Z axes are assumed positive
+    according to RHR.
+
+    Attributes: TODO
+        geometry (Union(SurfaceGeometry, CompoundGeometry)): The geometry of
+            the section.
+        name (str): The name of the section.
+        section_calculator (GenericSectionCalculator): The object responsible
+            for performing different calculations on the section (e.g. bending
+            strength, moment curvature, etc.).
+    """
+
+    def __init__(self, height: float, width: float, concrete: Concrete, name: t.Optional[str] = None, integrator: t.Literal['marin', 'fiber'] = 'marin') -> None:
```

Comment:
The material of a rectangular section is not concrete generally but any `Material`

Reply:
Fixed